File: src/main/java/de/fourofakind/businesshotel/server/entities/customers/ContactData.java
package de.fourofakind.businesshotel.server.entities.customers;

//

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.List;

/**
 * object to store personal data of customer
 */

@Entity(name="ContactData")
@Table(name="contact_data")
@AllArgsConstructor
@NoArgsConstructor
@Setter
@Getter
public class ContactData {

    //Attributes
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="contact_data_id")
    private Integer contactDataID;
    private String firstName;
    private String lastName;
    private String streetName;
    private String streetNumber;
    private String postalCode;
    private String cityName;
    private String phone;
    private String mailAddress;
    private String paymentCredentials;

    //Mappings
    @OneToMany(mappedBy = "contactData")
    private List<Customer> customers;

    //Constructors
    public ContactData(String firstName, String lastName, String streetName, String streetNumber, String postalCode, String cityName, String mailAddress)
    {
        this.firstName = firstName;
        this.lastName = lastName;
        this.streetName = streetName;
        this.streetNumber = streetNumber;
        this.postalCode = postalCode;
        this.cityName = cityName;
        this.mailAddress = mailAddress;
    }

    //Setter/Getter
    //managed by Lombok


}

File: src/main/java/de/fourofakind/businesshotel/client/src/app/services/contactdata/contactdata.ts
import {EmployeeSelf} from "../employee/employee";

export interface RawData {
  _embedded: Embedded;
  // _links: Links;
  page: Page;
}

export interface Page {
  size: number;
  totalElements: number;
  totalPages: number;
  number: number
}

export interface Embedded {
  contactdata: Contactdata[];
}

export interface Contactdata {
  contactDataID?:number;
  firstName: string;
  lastName: string;
  streetName: string;
  streetNumber: string;
  postalCode: string;
  cityName: string;
  phone: string;
  mailAddress: string;
  paymentCredentials?: string;
  _links?: ContactDataLinks;
}

export interface ContactDataLinks {
  self: Self;
  contactData: ContactDataSelf;
  customers: Customers;
}

export interface Self {
  href: string;
}

export interface ContactDataSelf {
  href: string;
}

export interface Customers {
  href: string;
}
